### DISCUSSION TITLE
Where to place image files?

### DISCUSSION THREAD

**Discussion by ben519** [squidfunk/mkdocs-material#3424] [Q&A] [answered]
Any advice on **where** to place image files which are referenced in my docs? For example, suppose I have a directory structure like

```
docs/
  assets/
  section1/
    tutorial1.md
    tutorial2.md
  section2/
    tutorial3.md
    tutorial4.md
site/
  assets/
```

and suppose I'm writing the docs for `tutorial2.md` which reference a few image files

*tutorial2.md*
```
# Tutorial 2

This is my tutorial..
![alt text](path/to/my-image.png)
```

Where exactly should `my-image.png` live my file structure? I noticed [in this doc article](https://squidfunk.github.io/mkdocs-material/reference/images/?h=images#light-and-dark-mode) you placed the image files in [`docs/assets/images/`](https://github.com/squidfunk/mkdocs-material/tree/5a96912cce06f78c679d69e95f59dd7a6a72993d/docs/assets/images). Is this location recommended for some reason?

**Comment by squidfunk** [comment, marked as answer]
As long as images are located in the `docs` folder (or any other folder set via `docs_dir`in `mkdocs.yml`), MkDocs will find them. Image files need to be linked the same way as internal links to other Markdown files. Material for MkDocs itself hosts some images in `docs/assets`, and images related to blog content in subfolders of the `blog` folder. It's entirely up to you.

**Reply by ben519** [reply]
Got it. Thanks!

**Reply by OndrejValenta** [reply]
Thank god. :-)
